Framework/PythonInterface/plugins/algorithms/HB3AFindPeaks.py

```diff
@@ -0,0 +1,164 @@
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+from mantid.api import AlgorithmFactory, IMDWorkspaceProperty, IPeaksWorkspaceProperty, PythonAlgorithm, PropertyMode
+from mantid.kernel import Direction, EnabledWhenProperty, PropertyCriterion, StringListValidator
+from mantid.simpleapi import DeleteWorkspace, FindPeaksMD, FindUBUsingIndexedPeaks, \
+    FindUBUsingLatticeParameters, IndexPeaks, ShowPossibleCells, SelectCellOfType, OptimizeLatticeForCellType, \
+    PredictFractionalPeaks
+
+
+class HB3AFindPeaks(PythonAlgorithm):
+
+    def category(self):
+        return "Crystal\\Corrections"
+
+    def seeAlso(self):
+        return ["FindPeaksMD", "OptimizeLatticeForCellType"]
+
+    def name(self):
+        return "HB3AFindPeaks"
+
+    def summary(self):
+        return 'Given a MD workspace in Q-space, calculates the UB matrix and finds peaks; the UB matrix can be ' \
+               'optionally refined using lattice parameters.'
+
+    def PyInit(self):
+        self.declareProperty(IMDWorkspaceProperty("InputWorkspace", defaultValue="", optional=PropertyMode.Mandatory,
+                                                  direction=Direction.Input),
+                             doc="Input MD workspace (in Q-space) to use for peak finding")
+
+        cell_type = StringListValidator()
+        cell_type.addAllowedValue("Cubic")
+        cell_type.addAllowedValue("Hexagonal")
+        cell_type.addAllowedValue("Rhombohedral")
+        cell_type.addAllowedValue("Tetragonal")
+        cell_type.addAllowedValue("Orthorhombic")
+        cell_type.addAllowedValue("Monoclinic")
+        cell_type.addAllowedValue("Triclinic")
+        self.declareProperty("CellType", "", cell_type, doc="Cell type to use for UB refining")
+
+        centering_type = StringListValidator()
+        centering_type.addAllowedValue("F")
+        centering_type.addAllowedValue("I")
+        centering_type.addAllowedValue("C")
+        centering_type.addAllowedValue("P")
+        centering_type.addAllowedValue("R")
+        self.declareProperty("Centering", "P", centering_type, doc="Centering to use for selecting cells")
+
+        # Some of the options to pass through to FindPeaksMD (options like CalculateGoniometerForCW, FlipX, etc are
+        # assumed to be True)
+        self.declareProperty("MaxPeaks", 1000, doc="Maximum number of peaks to find.")
+        self.declareProperty("PeakDistanceThreshold", 0.25, doc="Threshold distance for rejecting peaks that are found "
+                                                                "to be too close from each other")
+        self.declareProperty("DensityThresholdFactor", 50.0,
+                             doc="Scaling factor which the overall signal density will be multiplied by to determine "
+                                 "a threshold for determining peaks.")
+
+        self.declareProperty("Wavelength", defaultValue=1.558,
```

Comment:
I think this should be default empty otherwise it will suggest to the user that it is required.